Online-Messaging-App/src/app/home/header/header.component.ts

```diff
@@ -8,11 +10,26 @@ import { AuthenticationService } from "../../shared/authentication.service";
 })
 export class HeaderComponent implements OnInit {
     userLoggedIn = false;
+    user;
+    private usersAPI = APIConfig.usersAPI;
 
-    constructor(private auth: AuthenticationService) {
+    constructor(private auth: AuthenticationService, private http: HttpClient) {
         this.userLoggedIn = auth.isLoggedIn();
     }
 
     ngOnInit(): void {
+        if (this.userLoggedIn==true)
+        this.getUserInfo();
+    }
+
+    getUserInfo(): void {
```

Comment:
Ya you really don't need this. The current username is always available in memory, just need to get it from the authentication service

Reply:
done